services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -1386,6 +1385,8 @@ export class WorkflowService {
             } as InputJsonValue,
             config: merge(workflowDefinition.config, validatedConfig || {}) as InputJsonValue,
             status: 'active',
+            state: workflowDefinition.definition.initial,
```

Comment:
If makes sense, running the machine and getting the current state will be more reliable. Would give access to the current tag, too.

Reply:
Let's make sure we're not missing other instances of `workflowRuntimeDataRepository.create`.

Reply:
true, but the change is quite scary - I wanted to fix it here without changing the logic.
so, if we want to run the machine when creating it. we need to re-validate that all the schemas are valid from get-go

Reply:
Alright, let's just make sure it's documented somewhere. It could be a good candidate for Prisma middleware too. 🙂